src/main/cv/python/apriltag.py

```diff
@@ -97,32 +120,43 @@ def estimate_3d_pose(self, image, frame_ann, ARUCO_LENGTH_METERS):
                     tvec[2] =  original_z + AT_Z_OFFSET
 
                     original_x = tvec[0]
-                    tvec[0] =  original_x + AT_X_OFFSET
+                    tvec[0] =  (original_x + AT_X_OFFSET)
 
                     pose_list.extend(tvec)
-                    pose_list.extend(rvec)
+                    euler_rvec = self.rotation_vector_to_euler_angles(rvec)
+                    pose_list.extend(euler_rvec)
                     
-                    #print("tvec: ", tvec)
+                    # print("euler_rvec: ", euler_rvec)
                     self.draw_axis_on_image(frame_ann, self.camera_matrix, self.dist_coeffs, rvec, tvec, cvec, 0.1)
+            
+            pose_list = self.sort_tags_distance(pose_list)
 
             return pose_list
     
-    #returns the apriltag id of the apriltag closest to the center of the camera assuming that the camera is mounted at the center of the robot
-    def get_closest_tag(self, pose_list):
-        y_poses = {}
-        for i in range(len(pose_list)): 
-            translational_vector = pose_list[i][0]
-            y_poses[self.detectedIds[i]] = translational_vector
+    # sorts the tags in the list by their hypotenuse (sqrt of x^2 + z^2)
+    def sort_tags_distance(self, pose_list):
+        # pose list: [id, cvec, cvec, cvec, x, y, z, rvec, rvec, rvec, id, ...]
+        # x value = index of id + 4
+        # z value = index of id + 6
+        hyp_poses = {} # store dictionary in format of {id: hypotenuse}
+        for i in range(0, len(pose_list), 10): 
+            hyp = math.sqrt(pose_list[i + 4] ** 2 + pose_list[i + 6] ** 2) # hypotenuse = sqrt(x^2 + y^2)
+            hyp_poses[pose_list[i]] = hyp
+        
+        # sort hypotenuse dictionary
+        sorted_hyp_dict = dict(sorted(hyp_poses.items(), key=lambda item: item[1]))
```

Comment:
Reminder to fix this. dict does not maintain key ordering. If you want a specific iteration order, you should do something like `for id in sorted(hyp_poses.keys...)`.